test/hotspot/jtreg/serviceability/dcmd/vm/VMDebugTest.java

```diff
@@ -0,0 +1,228 @@
+        } else {
+            testFind(executor);
+        }
+    }
+
+    public void testFind(CommandExecutor executor) {
+        OutputAnalyzer output = executor.execute("VM.debug find");
+        output.shouldContain("missing argument");
+        // Find and test a thread id:
+        OutputAnalyzer threadPrintOutput = executor.execute("Thread.print");
+        BigInteger ptr = findPointer(threadPrintOutput, thread_id_line, 1);
+        output = executor.execute("VM.debug find " + pointerText(ptr));
+        output.shouldContain(" is a thread");
+        // verbose shows output like:
+        // "main" #1 [17235] prio=5 os_prio=0 cpu=1265.79ms elapsed=6.12s tid=0x000014e37802bd80 nid=17235 in Object.wait()  [0x000014e3817d4000]
+        //    java.lang.Thread.State: WAITING (on object monitor)
+        // Thread: 0x000014e37802bd80  [0x4353] State: _running _at_poll_safepoint 0
+        // ...
+        // Also a debug vm shows: JavaThread state: _thread_blocked
+        // ...
+        output = executor.execute("VM.debug find -verbose " + pointerText(ptr));
+        output.shouldContain("java.lang.Thread.State: WAITING");
+
+        // Known bad pointers:
+        output = executor.execute("VM.debug find 0x0");
+        output.shouldContain("address not safe");
+
+        output = executor.execute("VM.debug find -1");
+        output.shouldContain("address not safe");
+
+        // Find and test a Java Object:
+        threadPrintOutput = executor.execute("Thread.print");
+        ptr = findPointer(threadPrintOutput, waiting_on_mylock, 1);
+        output = executor.execute("VM.debug find " + pointerText(ptr));
+        System.out.println(output);
+        output.shouldContain(" is an oop: ");
+        output.shouldContain(" - ---- fields (total size");
+        // " - private 'myInt' 'I' @12  12345 (0x00003039)"
+        output.shouldContain(" - private 'myInt' 'I'");
+        output.shouldContain(" 12345 (");
+
+        BigInteger badPtr = ptr.add(BigInteger.ONE);
+        output = executor.execute("VM.debug find " + pointerText(badPtr));
+        output.shouldContain("misaligned");
+        badPtr = badPtr.add(BigInteger.ONE);
+        output = executor.execute("VM.debug find " + pointerText(badPtr));
+        output.shouldContain("misaligned");
+    }
+
+    public BigInteger findPointer(OutputAnalyzer output, Pattern pattern, int regexGroup) {
+        Iterator<String> lines = output.asLines().iterator();
+        Boolean foundMatch = false;
+        BigInteger ptr = null;
+        while (lines.hasNext()) {
+            String line = lines.next();
+            Matcher m = pattern.matcher(line);
+            if (m.find()) {
+                System.out.println("Matched line: " + line);
+                foundMatch = true;
+                String p = m.group(regexGroup);
+                ptr = new BigInteger(p, 16);
+                System.out.println("Found '" + pattern +"', using pointer: 0x" + ptr.toString(16));
```

Comment:
Would it be more logical to say this?
  Found pointer: 0x" + ptr.toString(16)'" + " for pattern " +pattern